mantle/kola/harness.go

```diff
@@ -875,6 +904,9 @@ func allTestsAllowRerunSuccess(testsBank map[string]*register.Test, testsToRerun
 	// Iterate over the tests that were re-ran. If any of them don't
 	// allow rerun success then just exit early.
 	for _, test := range testsToRerun {
+		if IsWarningOnFailure(test) {
+			continue
+		}
```

Comment:
do we want this?

Reply:
I think so. For example denylisted tagged `warn: true` tests shouldn't affect `allow-rerun-success tag=foo` failed tests.

Reply:
ahh I see now. if a warn:true test didn't have `tag=foo` then the loop would exit early because it found a test that didn't allow rerun success, but we don't want this to count against that.

Maybe the better thing to do here would be to not rerun warn:true tests. i.e. we'd filter it out in `getRerunnable(` somehow. I'm not sure if that's a great idea or not, but spitballing.

Reply:
i've removed this check and now `kola` simply reruns everything. Anyhow we sort `denylist + warn` out later